great_expectations/rule_based_profiler/parameter_builder/histogram_parameter_builder.py

```diff
@@ -158,48 +173,43 @@ def _build_parameters(
         Returns:
             Attributes object, containing computed parameter values and parameter computation details metadata.
         """
-        # Obtain bucketize_data directive from "rule state" (i.e., variables and parameters); from instance variable otherwise.
-        bucketize_data = get_parameter_value_and_validate_return_type(
-            domain=domain,
-            parameter_reference=self.bucketize_data,
-            expected_return_type=bool,
-            variables=variables,
-            parameters=parameters,
-        )
+        is_categorical: bool = not self._bucketize_data
 
-        is_categorical: bool = not bucketize_data
+        bins: Optional[MetricValue] = None
 
-        fully_qualified_column_partition_metric_single_batch_parameter_builder_name: str = f"{RAW_PARAMETER_KEY}{self._column_partition_metric_single_batch_parameter_builder_config.name}"
-        # Obtain "column.partition" from "rule state" (i.e., variables and parameters); from instance variable otherwise.
-        column_partition_parameter_node: ParameterNode = get_parameter_value_and_validate_return_type(
-            domain=domain,
-            parameter_reference=fully_qualified_column_partition_metric_single_batch_parameter_builder_name,
-            expected_return_type=None,
-            variables=variables,
-            parameters=parameters,
-        )
-        bins: MetricValue = column_partition_parameter_node[
-            FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY
-        ]
-
-        if bins is None:
-            is_categorical = True
-        elif not is_categorical:
-            ndarray_is_datetime_type: bool = is_ndarray_datetime_dtype(
-                data=bins,
-                parse_strings_as_datetimes=True,
+        if not is_categorical:
+            fully_qualified_column_partition_metric_single_batch_parameter_builder_name: str = f"{RAW_PARAMETER_KEY}{self._column_partition_metric_single_batch_parameter_builder_config.name}"
+            # Obtain "column.partition" from "rule state" (i.e., variables and parameters); from instance variable otherwise.
+            column_partition_parameter_node: ParameterNode = get_parameter_value_and_validate_return_type(
+                domain=domain,
+                parameter_reference=fully_qualified_column_partition_metric_single_batch_parameter_builder_name,
+                expected_return_type=None,
+                variables=variables,
+                parameters=parameters,
             )
-            bins_ndarray_as_float: MetricValue
-            if ndarray_is_datetime_type:
-                bins_ndarray_as_float = convert_ndarray_datetime_to_float_dtype(
-                    data=bins
+            bins = column_partition_parameter_node[
+                FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY
+            ]
+
+            if bins is None:
```

Comment:
Can you describe this case with an example? That may help me understand whether the logic could be simplified. 

In particular, I'm trying to understand the case where the user has asked for bucketized data, but we're returning invalid bins. Why isn't that just an error?

Reply:
@jcampbell I answered this comprehensively before, but am not seeing my answer (it had to do with the original spec in JIRA: GREAT-598).  I am looking for it.

Reply:
@jcampbell Let me try to repeat here what I wrote earlier (and now cannot find in GitHub): Per JIRA: GREAT-598, we have to support a case when we original request "column.histogram", but upon calculating the bins, we find that the bins value is incorrect, so we opt to request "column.partition" instead.  I also added the check if bins is NULL to this.  Furthermore, our "ColumnHistogram" class does not support the "datetime.datetime" type, so encountering such a type also switches from requesting "column.histogram" to "column.partition" ("ColumnPartition" does support "datetime.datetime").  Thanks.

Reply:
@alexsherstinsky I simply don't see that requirement spelled out in the ticket you mention. Critically, in this case, the parameter builder is intended to be invoked only after the domain has been selected in such a way that we would know whether to build bins for the data first.

I don't understand your reference to support for datetime types -- I would think that datetime types could be treated as either discrete or continuous as appropriate for the case at hand (though generally of course that would also correlate with cardinality).

So, I don't see a case for the complexity proposed here.